csrc/apis/c/CMakeLists.txt

```diff
@@ -4,38 +4,62 @@ project(capis)
 
 include(${CMAKE_SOURCE_DIR}/cmake/MMDeploy.cmake)
 
+set(TASK_LIST
+        common
+        model
+        executor
+        pipeline)
+
 if ("all" IN_LIST MMDEPLOY_CODEBASES)
-    set(TASK_LIST "classifier;detector;segmentor;text_detector;text_recognizer;"
-        "pose_detector;restorer;model;rotated_detector")
+    list(APPEND TASK_LIST
+            classifier
+            detector
+            segmentor
+            text_detector
+            text_recognizer
+            pose_detector
+            restorer
+            rotated_detector)
 else ()
-    set(TASK_LIST "model")
     if ("mmcls" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "classifier")
+        list(APPEND TASK_LIST classifier)
     endif ()
     if ("mmdet" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "detector")
+        list(APPEND TASK_LIST detector)
     endif ()
     if ("mmseg" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "segmentor")
+        list(APPEND TASK_LIST segmentor)
     endif ()
     if ("mmedit" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "restorer")
+        list(APPEND TASK_LIST restorer)
     endif ()
     if ("mmocr" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "text_detector")
-        list(APPEND TASK_LIST "text_recognizer")
+        list(APPEND TASK_LIST text_detector)
+        list(APPEND TASK_LIST text_recognizer)
     endif ()
     if ("mmpose" IN_LIST MMDEPLOY_CODEBASES)
-        list(APPEND TASK_LIST "pose_detector")
+        list(APPEND TASK_LIST pose_detector)
     endif ()
     if ("mmrotate" IN_LIST MMDEPLOY_CODEBASES)
         list(APPEND TASK_LIST "rotated_detector")
-    endif()
+    endif ()
 endif ()
 
 foreach (TASK ${TASK_LIST})
     set(TARGET_NAME mmdeploy_${TASK})
-    mmdeploy_add_library(${TARGET_NAME} ${TASK}.cpp)
+    mmdeploy_add_library(${TARGET_NAME} ${CMAKE_CURRENT_SOURCE_DIR}/${TASK}.cpp)
+    if (NOT TASK STREQUAL common)
```

Comment:
Cyclic dependencies among `mmdeploy_common`, `mmdeploy_executor`, `mmdeploy_model` and `mmdeploy_pipline` 
I think `${TARGET_NAME}` is trying to link `mmdeploy_common`, `mmdeploy_executor`, `mmdeploy_model` and `mmdeploy_pipline`, but the logic is not correct